src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2046,14 +2092,46 @@ static bool appendContextAttributeItem
 {
   if (legalIdUsage(attrs, targetAttr))
   {
-    appendAttribute(attrs, toSet, toPush, targetAttr, actualUpdate, apiVersion);
+    bool actualAppend = appendAttribute(attrs, toSet, toPush, targetAttr, actualUpdate, apiVersion);
     entityModified = actualUpdate || entityModified;
     updateAttrInNotifyCer(notifyCerP, targetAttr);
 
     /* Check aspects related with location */
-    if (targetAttr->getLocation().length() > 0)
+    std::string locationString = targetAttr->getLocation(apiVersion);
```

Comment:
The location stuff in this function has been heavily re-written. Hard to follow in diff mode. I'd recommend to have a look to the final code of this part directly.